### PR TITLE
[Merged by Bors] - refactor(CategoryTheory/MorphismProperty): turn `StableUnderBaseChange` into a class

### PR DESCRIPTION
**Pull request by chrisflav**
This allows for less tedious extra arguments.

--- 
A consistent renaming of `StableUnderBaseChange` to `IsStableUnderBaseChange` and `StableUnderCobaseChange` to `IsStableUnderCobaseChange` can be done in a follow-up PR.

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/AlgebraicGeometry/Morphisms/OpenImmersion.lean [hunk 1]
```diff
@@ -62,9 +62,9 @@ instance : IsLocalAtTarget (stalkwise (fun f ↦ Function.Bijective f)) := by
 instance isOpenImmersion_isLocalAtTarget : IsLocalAtTarget @IsOpenImmersion :=
   isOpenImmersion_eq_inf ▸ inferInstance
 
-theorem isOpenImmersion_stableUnderBaseChange :
+instance isOpenImmersion_stableUnderBaseChange :
```
[the remark this answers is not in the document]
Comment (chrisflav) [reply Mathlib/AlgebraicGeometry/Morphisms/OpenImmersion.lean:65]: Agreed. Maybe we should run a regular "AG maintenance party" where we just take care of some technical debt, e.g. split some files, eliminate some `erw`s, fix naming :D

**Comment by erdOne** [comment]
Thanks for doing this!
maintainer merge

File: Mathlib/AlgebraicGeometry/Morphisms/OpenImmersion.lean (hunk 1 above)
[the remark this answers is not in the document]
Comment (erdOne) [reply Mathlib/AlgebraicGeometry/Morphisms/OpenImmersion.lean:65]: Sure. Will there be anyone other than you & me though 😅

[the remark this answers is not in the document]
Comment (chrisflav) [reply Mathlib/AlgebraicGeometry/Morphisms/OpenImmersion.lean:65]: Maybe if we do some advertisement?

**Comment by joelriou** [comment]
Thanks very much!

bors merge

**Pull request closed without merging**